components/sbm-recipes-mule-to-boot/src/main/java/org/springframework/sbm/mule/actions/JavaDSLAction2.java

```diff
@@ -90,16 +91,15 @@ private void handleTopLevelElements(BuildFile buildFile, MuleMigrationContext mu
             } else {
                 definitionSnippet = new UnknownTopLevelElement(tle);
             }
-            buildFile.addDependencies(buildDependencies(definitionSnippet));
+            dependencies.addAll(buildDependencies(definitionSnippet));
             flowConfigurationSource.getType().addMethod(definitionSnippet.renderDslSnippet(), definitionSnippet.getRequiredImports());
         }
+        buildFile.addDependencies(dependencies);
```

Comment:
This needs to be dine before the call to `addMethod()`, otherwise types will not be resolved